src/Symfony/Component/Workflow/Workflow.php

```diff
@@ -79,25 +81,28 @@ public function getMarking($subject)
     /**
      * {@inheritdoc}
      */
-    public function can($subject, $transitionName)
+    public function can($subject, $transitionName, bool $throwException = false)
```

Comment:
Imho $throwException should be default true on next major. A deprecation should be added for calling ->can with $throwException = false. keeping this bool around for longer than necessary is bloating the API imho.

Reply:
I disagree with that ;) IMHO knowing exactly why a transition is blocked is not so common.
more over, in twig it will not work well.

Reply:
Hm. Imho Every user driven workflow wants to know exactly why a transition isnt enabled. And also what transitions can be enabled in a given state (transition availibility). For highly automated workflows i can See your point.

To Clarify my desired behavior: `can` s‭hould always Return a bool except when a transition isnt even defined in the workflow. A second method should always return a blocker list (except for undefined transitions). A third method `isTransitionDefined` That always returns bool and Never throws.

Reply:
> To Clarify my desired behavior: can s‭hould always Return a bool except when a transition isnt even defined in the workflow. A second method should always return a blocker list (except for undefined transitions). A third method isTransitionDefined That always returns bool and Never throws.

Why not ;) What do other think ?

Reply:
But I have two concerns with your proposal:

> : can s‭hould always Return a bool except when a transition isnt even defined in the workflow

This is a BC break

> A second method should always return a blocker list (except for undefined transitions)

What would be the name

Reply:
A) keep the throwException param with default false, But deprecate Passing false 

B) Workflow::whyNot ::checkTransition ::testTransition ::determineTransitionAvailibility (Probably the 2nd is my favorite)

Reply:
Ok, I will do that with `checkTransition` (I'm not really fan of this name though )

Reply:
If you think another name is better then go for it. I myself am not convinced 100% by checkTransition

Reply:
I did not added `isTransitionDefined`. IMHO it's useless. If you really think it's useful please open an issue and let's discuss it there. 

I would like to merge this PR ASAP ;)